Index: docker-compose.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/docker-compose.yml b/docker-compose.yml
new file mode 100644
--- /dev/null	(date 1687917301562)
+++ b/docker-compose.yml	(date 1687917301562)
@@ -0,0 +1,22 @@
+version: "3.8"
+
+services:
+  angular-tutorial:
+    container_name: angular
+    tty: true
+    init: true
+    build:
+      context: ./
+      dockerfile: ./Dockerfile.dev
+    ports:
+      - '4242:4200'
+    volumes:
+      - ./:/usr/src:cached
+
+    environment:
+      - TZ=Asia/Tokyo
+      # Angular ng build の確認メッセージスキップ
+      - NG_CLI_ANALYTICS=false
+      - NODE_ENV=development
+      - TINI_SUBREAPER=true
+      - NODE_OPTIONS=--max-old-space-size=8192
Index: .gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.gitignore b/.gitignore
new file mode 100644
--- /dev/null	(date 1687917144650)
+++ b/.gitignore	(date 1687917144650)
@@ -0,0 +1,2 @@
+.angular
+node_modules
\ No newline at end of file
Index: Dockerfile.dev
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Dockerfile.dev b/Dockerfile.dev
new file mode 100644
--- /dev/null	(date 1687917238208)
+++ b/Dockerfile.dev	(date 1687917238208)
@@ -0,0 +1,16 @@
+FROM node:18.10.0-slim as node
+
+FROM ubuntu:22.04
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    tini \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=node /usr/local/include/ /usr/local/include/
+COPY --from=node /usr/local/lib/ /usr/local/lib/
+COPY --from=node /usr/local/bin/ /usr/local/bin/
+
+WORKDIR /usr/src
+ENV PATH=/usr/src/node_modules/.bin:$PATH
